packages/app/ui/components/BigInput.tsx

```diff
@@ -261,11 +284,90 @@ export function BigInput({
     setShowAttachmentSheet(false);
   }, []);
 
+  const handleInlineImageSelect = useCallback(
+    async (assets: ImagePickerAsset[]) => {
+      if (assets.length > 0 && editorRef.current?.editor) {
+        const asset = assets[0];
+
+        // For inline images in notebooks, we need to upload the image first
+        // then insert the uploaded URL into the editor
+        // IMPORTANT: We do NOT use uploadAssets because that adds to attachments
+        // Instead, we upload directly and wait for the URL
+        try {
+          // Upload the image directly without adding to attachments
+          await uploadAssetToStorage(asset, true);
+
+          if (!isMountedRef.current) return;
+
+          // Wait for the upload to complete and get the S3 URL
+          const uploadStates = await waitForUploads([asset.uri]);
+
+          // Check again after await
+          if (!isMountedRef.current) return;
+
+          const uploadState = uploadStates[asset.uri];
+
+          if (uploadState?.status === 'success' && editorRef.current?.editor) {
+            // Insert the S3 URL into the editor
+            const s3Url = uploadState.remoteUri;
+            (editorRef.current.editor as any).setImage(s3Url);
+          } else if (isMountedRef.current) {
+            logger.error('Failed to upload image, upload state:', uploadState);
```

Comment:
can we track these errors? And have something about it being an inline notebook image in the message